### DISCUSSION TITLE
Vercel Function sometimes is very very slow for server side pages

### DISCUSSION THREAD

**Discussion by akshitkrnagpal** [vercel/next.js#30776] [Help]
I have an app with a server side rendered page. 

Sometimes the edge function of that page takes outrageous time. There is nothing cached in the `getServerSideProps` so it shouldn't be any kind of cache (at least on our end). I have noticed it mostly happens when there is `Init Duration` also present (which I assume is cold start of serverless function).

Any idea how we can Improve it?

```
[GET] /{url-redacted}.                 10:47:51:89
Function Status: 200
Edge Status:     200
Duration:        10891.90 ms
Init Duration:   449.21 ms
Memory Used:     269 MB

[GET] /{url-redacted}.                 10:47:49:99
Function Status: 200
Edge Status:     200
Duration:        29.12 ms
Init Duration:   N/A
Memory Used:     258 MB
```

**Comment by FabianL1** [comment]
I also have that problem. Any solution to it?

**Reply by icyJoseph** [reply]
What kind of modules do you import on this function? I remember that there used to be a library that imported a bunch of JSON files, whenever it was brought into scope, and that made parsing extremely time consuming. I would start looking there.

**Comment by benevbright** [comment]
the same issue. fast enough most of requests but sometimes 10~15 sec randomly.
